packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -59,6 +72,86 @@ export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultPr
     },
     [setOpen],
   );
+  state.on.length === 0 && state.on.push('click');
+
+  useMenuPopup(state);
+  useOnClickOutside({ element: document, refs: [state.menuPopupRef, triggerRef], callback: () => setOpen(false) });
+  useOnEscape({ element: document, callback: () => setOpen(false) });
+  return state;
+};
+
+const useMenuPopup = (state: MenuState) => {
+  // TODO use Popper for the popup ref
+  const menuPopupRef = React.useRef<HTMLElement>(null) as React.MutableRefObject<HTMLElement>;
+  state.menuPopupRef = menuPopupRef;
+  state.menuPopup.children = (Component, originalProps) => {
+    const newProps = { 'aria-labelledBy': state.triggerId, ...originalProps };
+
+    if (state.on.includes('hover')) {
+      newProps.onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
+        state.setOpen(true);
+        originalProps?.onMouseEnter?.(e);
+      };
+      newProps.onMouseLeave = (e: React.MouseEvent<HTMLElement>) => {
+        state.setOpen(false);
+        originalProps?.onMouseLeave?.(e);
+      };
+    }
+
+    newProps.onBlur = (e: React.FocusEvent<HTMLElement>) => {
+      state.setOpen(false);
+      originalProps?.onBlur?.(e);
+    };
+
+    return (
+      // Shorthand render funct types have issues
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      <Component {...newProps} ref={state.menuPopupRef}>
+        {state.menuList}
+      </Component>
+    );
+  };
 
   return state;
 };
+
+// TODO handle nested menus
+const useOnClickOutside = (options: {
+  element?: Node | Window | Document;
```

Comment:
_Another note_: probably this should not be configurable and we should just use `document` from `useFluent` as I don't have any other case in my head when this should an another element. Do you have an example when it can be useful?

For example:
- https://github.com/Andarist/use-onclickoutside/blob/master/src/index.ts
- https://usehooks.com/useOnClickOutside/

Reply:
the option itself is optional, but I see where you're going with this. 

I'll keep the possibility of configuration but call `useFluent` in this hook as the default

Reply:
keeping as is, since this is now in `react-utilities` and I'm not sure how we want to handle that dependnecy 😨